README.md

```diff
@@ -1,24 +1,82 @@
-# Node.js template for the AsyncAPI Generator
+<h5 align="center">
+  <br>
+  <a href="https://www.asyncapi.org"><img src="https://github.com/asyncapi/parser-nodejs/raw/master/assets/logo.png" alt="AsyncAPI logo" width="200"></a>
+  <br>
+  AsyncAPI Node.js Template
+</h5>
 
-## Usage
+![npm](https://img.shields.io/npm/v/@asyncapi/nodejs-template?style=for-the-badge) ![npm](https://img.shields.io/npm/dt/@asyncapi/nodejs-template?style=for-the-badge)
 
-```
-ag asyncapi.yaml @asyncapi/nodejs-template -o output -p server=production
-```
+- [Overview](#overview)
+- [Supported protocols](#supported-protocols)
+- [How to use the template](#how-to-use-the-template)
+    * [CLI](#cli)
+- [Specification requirements](#specification-requirements)
+- [Template configuration](#template-configuration)
+- [Custom hooks that you can disable](#custom-hooks-that-you-can-disable)
 
-If you don't have the AsyncAPI Generator installed, you can install it like this:
 
-```
+## Overview
+
+This template generates a Node.js application with any of the [supported protocols](#supported-protocols) endpoint, based on [Hermesjs](https://github.com/hitchhq/hermes).
+
+Other files are for the setup of developer environment, like `.editorconfig` or `.eslint`.
+
+## Supported protocols
+
+* [AMQP](https://en.wikipedia.org/wiki/Advanced_Message_Queuing_Protocol)
+* [MQTT](https://en.wikipedia.org/wiki/MQTT)
+* [Kafka](https://en.wikipedia.org/wiki/Apache_Kafka)
+* [WebSocket](https://en.wikipedia.org/wiki/WebSocket)
+
+
+
+## How to use the template
+
+This template must be used with the AsyncAPI Generator. You can find all available options [here](https://github.com/asyncapi/generator/).
+
+### CLI
+
+```bash
+# Install the AsyncAPI Generator
 npm install -g @asyncapi/generator
+
+# copy your asyncapi.yaml file to the current directory or create a new file.
+
+# Run generation
+ag asyncapi.yaml @asyncapi/nodejs-template -o output -p server=production
+##
+## Start the server 
+##
+
+# Go to the generated server
+cd output
+
+# Build generated application
+npm i
+
+# Start server
+npm start
```

Comment:
consider adding here instructions about installing mqtt client with proper description that it is just to quickly see the template in action

```bash
npm install mqtt -g
mqtt pub -t 'light/measured' -h 'test.mosquitto.org' -m '{"id": 1, "lumens": 3, "sentAt": "2017-06-07T12:34:32.000Z"}'
```